nomad/structs/structs.go

```diff
@@ -4736,6 +4740,8 @@ type TaskGroup struct {
 
 	// Volumes is a map of volumes that have been requested by the task group.
 	Volumes map[string]*VolumeRequest
+
+	ShutdownDelay *time.Duration
```

Comment:
This doesn't need to be a pointer as there's no difference between `nil` and `0`.

Reply:
This needed to be a pointer so that a job diff knew if it was set or not, otherwise I was getting 

```
Failed
=== RUN   TestJobDiff
--- FAIL: TestJobDiff (0.00s)
    diff_test.go:1196: case 19: got:
        Job "" (Edited):
        Group "bam" (Added):
        "Count" (Added): "" => "1"
        "ShutdownDelay" (Added): "" => "0"
```